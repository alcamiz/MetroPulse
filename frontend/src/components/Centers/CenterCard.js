import React from "react";
import { Link } from "react-router-dom";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";
import CenterPlaceholder from "../../placeholders/center_placeholder.png";

const styles = {
  link: {
      textDecoration: 'none',
  },
  card: {
      width: '20rem',
      height: 'auto'
  }
};

function CenterCard({ center }) {
  console.log(center);

  return (
    <Link to={`/test/${center.id_t}`} style={styles.link}>
      <Card style={styles.card}>
        <Card.Img variant="top" src={center.static_map_url} />
        <Card.Body>
          <Card.Title>{center.name}</Card.Title>
          <Card.Text><strong>Borough:</strong> {center.borough}</Card.Text>
          <Card.Text>Council Number: {center.council}</Card.Text>
          <Card.Text>Zipcode: {center.zip_code}</Card.Text>
          <Card.Text>Nearby Medical Facilities: {center.nearby_hospitals.length}</Card.Text>
        </Card.Body>
      </Card>
    </Link>
  );
}

export default CenterCard;
