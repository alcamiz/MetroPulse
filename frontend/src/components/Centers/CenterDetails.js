import React from "react";
import "../../instances/Instance.css";
import CenterPlaceholder from "../../placeholders/center_placeholder.png"; // ADD PLACEHOLDER IMAGE

import background from '../../shared/img/test_back.jpeg';
import '../../instances/Instance.css';
import Accordion from 'react-bootstrap/Accordion';
import Table from 'react-bootstrap/Table';
import { Link } from 'react-router-dom';


function CenterInfo({ center }) {
  return (
    <Accordion defaultActiveKey="0">
      <Accordion.Item eventKey="0">
        <Accordion.Header>Contact Information</Accordion.Header>
        <Accordion.Body>
          <Table striped bordered hover>
            <tbody>
              <tr>
                <td>Phone Number</td>
                <td>{center.phone}</td>
              </tr>
              <tr>
                <td>Address</td>
                <td>{center.address}, {center.nta_name}, NY {center.zip_code}</td>
              </tr>
              <tr>
                <td>Neighborhood</td>
                <td><Link to={`/hoods/${center.parent_neighborhood}`}>{center.nta_name}</Link></td>
              </tr>
              <tr>
                <td>Borough</td>
                <td>{center.borough}</td>
              </tr>
              <tr>
                <td>Council District</td>
                <td>{center.council}</td>
              </tr>
              <tr>
                <td>Instructions</td>
                <td>{center.howto}</td>
              </tr>
            </tbody>
          </Table>
        </Accordion.Body>
      </Accordion.Item>
      <Accordion.Item eventKey="1">
        <Accordion.Header>Nearby Medical Facilities</Accordion.Header>
        <Accordion.Body>
          <ul>
            { (center.nearby_hospitals == null)? 0 : center.nearby_hospitals.map((hospital) => (
              <li key={hospital.id_t}>
                <Link to={`/medical/${hospital.id_t}`}>{hospital.name}</Link>
              </li>
            )) }
          </ul>
        </Accordion.Body>
      </Accordion.Item>
    </Accordion>
  );
}

function CenterDetails({ center }) {
  return (
    <div className="Instance">
      <div class="container">
        <div class="row">
          <div class="col-sm">
            <header className="Instance-header">
              <img src={center.static_map_url} className="Instance-logo" alt="logo" /> 
              <h3 className="Instance-title">
                {center.name}
              </h3>
              <CenterInfo center={center} />
            </header>
          </div>
          <div class="col-sm">
            <div className="Instance-images">
              <img style={{ display: `block` }} width="400" height="400" src={center.image_url || "https://media.istockphoto.com/id/472149497/photo/goofy-pharmacist.jpg?s=612x612&w=0&k=20&c=OBV_lo-TKcZD3DwuYa23sVGvFeT8IsSNBjjfZbtWuts="} className="Instance-logo" alt="logo" />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default CenterDetails;
