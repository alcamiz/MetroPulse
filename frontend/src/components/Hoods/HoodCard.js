import React from "react";
import { Link } from "react-router-dom";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";
import HoodPlaceholder from "../../placeholders/neighborhood_placeholder.png"; 

const styles = {
  link: {
      textDecoration: 'none',
  },
  card: {
      width: '20rem',
      height: 'auto'
  }
};

function HoodCard( {hood} ) {
  // center

  return (
    <Link to={`/hoods/${hood.id_t}`} style={styles.link}>
      <Card style={styles.card}>
        <Card.Img variant="top" src={hood.static_map_url} />
        <Card.Body>
          <Card.Title>{hood.nta_name}</Card.Title>
          <Card.Text>Borough: {hood.borough}</Card.Text>
          <Card.Text>Population: {hood.population}</Card.Text>
          <Card.Text>County Code: {hood.fips_county_code}</Card.Text>
          <Card.Text>NTA Code: {hood.nta_code}</Card.Text>
          <Card.Text>Nearby Hospitals: { (hood.nearby_hospitals == null)? 0 : hood.nearby_hospitals.length}</Card.Text>
        </Card.Body>
      </Card>
    </Link>
  );
}

export default HoodCard;
