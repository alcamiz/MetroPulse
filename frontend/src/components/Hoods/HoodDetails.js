import React from "react";

import '../../instances/Instance.css';
import HoodPlaceholder from "../../placeholders/neighborhood_placeholder.png";

import background from '../../shared/img/test_back.jpeg';
import Accordion from 'react-bootstrap/Accordion';
import Table from 'react-bootstrap/Table';
import { Link } from 'react-router-dom';


function HoodInfo({ hood }) {
  return (
    <Accordion defaultActiveKey="0">
    <Accordion.Item eventKey="1">
      <Accordion.Header>Description</Accordion.Header>
      <Accordion.Body>
        <p>{hood.desc}</p>
      </Accordion.Body>
    </Accordion.Item>
    <Accordion.Item eventKey="2">
      <Accordion.Header>Information</Accordion.Header>
      <Accordion.Body>
        <Table striped bordered hover>
          <tbody>
            <tr>
              <td>Borough</td>
              <td>{hood.borough}</td>
            </tr>
            <tr>
              <td>Population</td>
              <td>{hood.population}</td>
            </tr>
            <tr>
              <td>County Code</td>
              <td>{hood.flips_county_code}</td>
            </tr>
            <tr>
              <td>NTA Code</td>
              <td>{hood.nta_code}</td>
            </tr>
          </tbody>
        </Table>
      </Accordion.Body>
    </Accordion.Item>
    <Accordion.Item eventKey="3">
      <Accordion.Header>Nearby Testing Centers</Accordion.Header>
      <Accordion.Body>
        <ul>
          {hood.nearby_centers.map((center) => (
            <li key={center.id_t}>
              <Link to={`/test/${center.id_t}`}>{center.name}</Link>
            </li>
          ))}
        </ul>
      </Accordion.Body>
    </Accordion.Item>
    <Accordion.Item eventKey="4">
      <Accordion.Header>Nearby Hospitals</Accordion.Header>
      <Accordion.Body>
        <ul>
          {hood.nearby_hospitals.map((hospital) => (
            <li key={hospital.id_t}>
              <Link to={`/medical/${hospital.id_t}`}>{hospital.name}</Link>
            </li>
          ))}
        </ul>
      </Accordion.Body>
    </Accordion.Item>
  </Accordion>
  );
}

function HoodDetails({hood}) {
  return (
    <div className="Instance" style={{backgroundImage: `url(${background})`}}>
      <header className="Instance-header">
          <img src={hood.static_map_url || HoodPlaceholder} className="Instance-logo" alt="logo" />  {/*FIX MAP URL */}
        <h3 className="Instance-title">
          {hood.nta_name}
        </h3>
        <HoodInfo hood={hood}/>
      </header>
      <div className="Instance-images">
        <img style={{display: `block`}} src={hood.image_url || "https://www.elikarealestate.com/blog/wp-content/uploads/2015/09/village-streets-nyc.jpg"} className="Instance-logo" alt="logo" />
      </div>
    </div>
  );
}

export default HoodDetails;
