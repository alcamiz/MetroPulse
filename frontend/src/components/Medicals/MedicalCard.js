import React from "react";
import { Link } from "react-router-dom";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";
import MedicalPlaceholder from "../../placeholders/medical_placeholder.png";
import {highlightWords} from "../../utils/highlightWords"

const styles = {
  link: {
      textDecoration: 'none',
  },
  card: {
      width: '20rem',
      height: 'auto'
  }
};

function MedicalCard({ medical, highlight }) {

  return (
    <Link to={`/medical/${medical.id_t}`} style={styles.link}>
    <Card style={styles.card}>
      <Card.Img variant="top" width="400" height="400" src={medical.static_map_url || "https://www.google.com/maps/d/thumbnail?mid=1rN8fFwktkFQlwWkL8mlSoximF6E"} />
      <Card.Body>
        <Card.Title dangerouslySetInnerHTML={{__html: highlightWords(medical.name, highlight)}}/>
        <Card.Text>Borough: {medical.borough}</Card.Text>
        <Card.Text>Council District: {medical.council_district}</Card.Text>
        <Card.Text>Zipcode: {medical.zip_code}</Card.Text>
        <Card.Text>Nearby Testing Centers: {(medical.nearby_centers == null) ? 0 : medical.nearby_centers.filter((center, index, self) => {
              return self.findIndex((c) => c.name === center.name) === index;
            }).length}</Card.Text>
      </Card.Body>
    </Card>
    </Link>
  );
}

export default MedicalCard;
