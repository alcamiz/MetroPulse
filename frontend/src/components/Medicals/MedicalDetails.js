import React from "react";
import "../../instances/Instance.css";
import MedicalPlaceholder from "../../placeholders/medical_placeholder.png"; 

import background from '../../shared/img/test_back.jpeg';
import '../../instances/Instance.css';
import Accordion from 'react-bootstrap/Accordion';
import Table from 'react-bootstrap/Table';
import { Link } from 'react-router-dom';


function MedicalInfo({ medical }) {
  return (
    <Accordion defaultActiveKey="0">
      <Accordion.Item eventKey="0">
        <Accordion.Header>Contact Information</Accordion.Header>
        <Accordion.Body>
          <Table striped bordered hover>
            <tbody>
              <tr>
                <td>Phone Number</td>
                <td>{medical.phone}</td>
              </tr>
              <tr>
                <td>Address</td>
                <td>{medical.address}</td>
              </tr>
              <tr>
                <td>Neighborhood</td>
                <td><Link to={'/hoods/' + medical.parent_neighborhood}>{medical.nta}</Link></td>
              </tr>
              <tr>
                <td>Borough</td>
                <td>{medical.borough}</td>
              </tr>
              <tr>
                <td>Council District</td>
                <td>{medical.council}</td>
              </tr>
            </tbody>
          </Table>
        </Accordion.Body>
      </Accordion.Item>
      <Accordion.Item eventKey="1">
        <Accordion.Header>Nearby Testing Centers</Accordion.Header>
        <Accordion.Body>
          <ul>
            {medical.nearby_centers.map((center) => (
              <li key={center.id_t}>
                <Link to={`/center/${center.id_t}`}>{center.name}</Link>
              </li>
            ))}
          </ul>
        </Accordion.Body>
      </Accordion.Item>
    </Accordion>
  );
}

function MedicalDetails({medical}) {
  return (
    <div className="Instance">
      <div class="container">
        <div class="row">
          <div class="col-sm">
            <header className="Instance-header">
              <img src={medical.static_map_url || "https://www.google.com/maps/d/thumbnail?mid=1rN8fFwktkFQlwWkL8mlSoximF6E"} className="Instance-logo" alt="logo" /> 
              <h3 className="Instance-title">
                {medical.facility_name}
              </h3>
              <MedicalInfo medical={medical}/>
            </header>
          </div>
          <div class="col-sm">
            <div className="Instance-images">
              <img style={ {display: `block`}} width="400" height="400" src={medical.image_url || "https://s3-prod.modernhealthcare.com/s3fs-public/styles/width_792/public/rural-health2_i.png"} className="Instance-logo" alt="logo" />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default MedicalDetails;
