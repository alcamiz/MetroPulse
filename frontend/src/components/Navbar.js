import React, { useState } from 'react';
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { Navbar, Nav, Form, FormControl, Button } from 'react-bootstrap';

function NavBar() {
  const [isSearchMode, setIsSearchMode] = useState(false);
  const [searchValue, setSearchValue] = useState('');

  const enterSearchMode = () => {
      setIsSearchMode(true);
  };

  const exitSearchMode = () => {
      setIsSearchMode(false);
      setSearchValue('');
  };

  const handleSearchChange = (e) => {
      setSearchValue(e.target.value);
  };

  return (
    <Navbar bg="dark" variant="dark" expand="lg" data-testid="navbar">
      <Navbar.Brand style={{ marginLeft: '20px' }} href="/">MetroPulse</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        {isSearchMode ? (
          <div className="mr-auto" style={{ flex: 1, display: 'flex', alignItems: 'center' }}>
            <FormControl
              type="text"
              placeholder="Search"
              className="mr-sm-2"
              value={searchValue}
              onChange={handleSearchChange}
              onBlur={exitSearchMode}
            />
            <Button variant="outline-info" onClick={exitSearchMode}>Close</Button>
          </div>
        ) : (
          <div style={{ display: 'flex', alignItems: 'center', width: '100%' }}>
            <Nav className="mr-auto">
              <Nav.Link href="/">Home</Nav.Link>
              <Nav.Link href="/medical/">Medical Facilities</Nav.Link>
              <Nav.Link href="/test/">Test Centers</Nav.Link>
              <Nav.Link href="/hoods/">Neighborhoods</Nav.Link>
              <Nav.Link href="/vis/">Visualizations</Nav.Link>
              <Nav.Link href="/prov/">Provider Visualizations</Nav.Link>
              <Nav.Link href="/about/">About</Nav.Link>
            </Nav>
          </div>
        )}
      </Navbar.Collapse>
    </Navbar>
  );
}

export default NavBar;
