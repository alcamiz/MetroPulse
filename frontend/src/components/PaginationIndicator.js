import React from "react";
import "../styles/Pagination.css"

const max = (val1, val2) => {
  return val1 > val2 ? val1 : val2;
};

const min = (val1, val2) => {
  return val1 < val2 ? val1 : val2;
};

const VALUES_BEFORE_AND_AFTER = 5;

const PaginationIndicator = ({ numPages, currPage, setCurrPage }) => {
  const valuesAround = [];
  for (
    let i = max(
      currPage -
        VALUES_BEFORE_AND_AFTER -
        (max(currPage + VALUES_BEFORE_AND_AFTER, numPages) - numPages),
      1
    );
    i < currPage;
    i++
  ) {
    valuesAround.push(i);
  }
  valuesAround.push(currPage);
  for (
    let i = currPage + 1;
    i <=
    min(
      currPage +
        VALUES_BEFORE_AND_AFTER -
        min(currPage - VALUES_BEFORE_AND_AFTER, 0),
      numPages
    );
    i++
  ) {
    valuesAround.push(i);
  }

  const selectNextButton = () => {
    setCurrPage(currPage + 1);
  };

  const selectPrevButton = () => {
    setCurrPage(currPage - 1);
  };

  const selectButton = (pageNumber) => {
    if (currPage !== pageNumber) {
      setCurrPage(pageNumber);
    }
  };

  return (
    <div className="pagination-indicator">
      {currPage !== 1 && (
        <button
          className="pagination-button prev-next-pagination-button"
          onClick={selectPrevButton}
        >
          {"<"}
        </button>
      )}
      {valuesAround.map((currVal) => {
        return (
          <button
            key={currVal}
            className={
              "pagination-button" +
              (currVal === currPage ? " pagination-selected" : "")
            }
            onClick={() => selectButton(currVal)}
          >
            <span>{currVal}</span>
          </button>
        );
      })}
      {currPage !== numPages && (
        <button
          className="pagination-button prev-next-pagination-button"
          onClick={selectNextButton}
        >
          {">"}
        </button>
      )}
    </div>
  );
};

export default PaginationIndicator;
