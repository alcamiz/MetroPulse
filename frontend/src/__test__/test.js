/**
* @jest-environment jsdom
*/

import React from "react"
import renderer from "react-test-renderer"
import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect';

import { BrowserRouter as Router, Route, Routes } from "react-router-dom"

import HoodCard from "../components/Hoods/HoodCard.js"
import CenterCard from "../components/Centers/CenterCard.js"
import MedicalCard from "../components/Medicals/MedicalCard.js"
import MedicalDetails from "../components/Medicals/MedicalDetails.js"
import HoodDetails from "../components/Hoods/HoodDetails.js"
import CenterDetails from "../components/Centers/CenterDetails.js"
import Navbar from "../components/Navbar.js"
import Loading from "../components/Loading.js"
import medical_test_json from "./medical_test.json"
import hood_test_json from "./hood_test.json"
import center_test_json from "./center_test.json"
import PaginationIndicator from "../components/PaginationIndicator.js";

const mockPerformanceMark = jest.fn();
window.performance.mark = mockPerformanceMark;

test('test hood card', () => {
    const hood = hood_test_json;
    const tree = renderer.create(
        <Router>
            <HoodCard
                hood={hood}
            />
        </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
})

test('test medical card', () => {
    const medical = medical_test_json;
    const tree = renderer.create(
        <Router>
            <MedicalCard
                medical={medical}
            />
        </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
})

test('test center card', () => {
    const center = center_test_json;
    const tree = renderer.create(
        <Router>
            <CenterCard
                center={center}
            />
        </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
})

test('test hood details', () => {
    const hood = hood_test_json;
    const tree = renderer.create(
        <Router>
            <HoodDetails
                hood={hood}
            />
        </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
})

test('test medical details', () => {
    const medical = medical_test_json;
    const tree = renderer.create(
        <Router>
            <MedicalDetails
                medical={medical}
            />
        </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
})

test('test center details', () => {
    const center = center_test_json;
    const tree = renderer.create(
        <Router>
            <CenterDetails
                center={center}
            />
        </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
})

test("Test whether the Navbar works", () => {
    const tree = renderer.create(
        <Router>
            <Navbar />
        </Router>
    )
    expect(tree).toMatchSnapshot()
})

test("Test the Navbar", async () => {
    render(
        <Router>
            <Navbar />
        </Router>
    );
    const navbar = screen.getByTestId("navbar");
    expect(navbar).toHaveTextContent("About");
    expect(navbar).toHaveTextContent("Home");
    expect(navbar).toHaveTextContent("Neighborhoods");
    expect(navbar).toHaveTextContent("Medical Facilities");
    expect(navbar).toHaveTextContent("Test Centers");
});

test("Test Loading", async () => {
    const tree = renderer.create(
        <Loading />
    )
    expect(tree).toMatchSnapshot()
})

test("Test Indicator", async () => {
    const tree = renderer.create(
        <PaginationIndicator />
    )
    expect(tree).toMatchSnapshot()
})
