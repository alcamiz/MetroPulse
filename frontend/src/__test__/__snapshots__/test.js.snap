// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test whether the Navbar works 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    style={
      Object {
        "marginLeft": "20px",
      }
    }
  >
    MetroPulse
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "width": "100%",
        }
      }
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="/"
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/medical/"
          disabled={false}
          href="/medical/"
          onClick={[Function]}
        >
          Medical Facilities
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/test/"
          disabled={false}
          href="/test/"
          onClick={[Function]}
        >
          Test Centers
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/hoods/"
          disabled={false}
          href="/hoods/"
          onClick={[Function]}
        >
          Neighborhoods
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/about/"
          disabled={false}
          href="/about/"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
  </div>
</nav>
`;
